packages/firebase_auth/ios/Classes/FirebaseAuthPlugin.m

```diff
@@ -30,7 +30,10 @@ - (FlutterError *)flutterError {
 
 @interface FLTFirebaseAuthPlugin ()
 @property(nonatomic, retain) NSMutableDictionary *authStateChangeListeners;
-@property(nonatomic, retain) FlutterMethodChannel *channel;
+@property(nonatomic, retain) FlutterMethodChannel *methodChannel;
+@property(nonatomic, retain) FlutterEventChannel *phoneAuthEventChannel;
+@property(nonatomic, copy) FlutterEventSink phoneAuthEventSink;
```

Comment:
Event streams seem a bit heavy for just emitting a single event. If there are no other uses for the stream, maybe just use a method call in the iOS to Dart direction?

Reply:
I also think the event stream setup is overkill for this, but I did it because [the Android implementation](ae74f4c4532ff796ebfb8b54d2adc8079c49202a) used it. Android's phone number auth seems to support a lot more features than iOS' implementation does. I just looked through that linked implementation and tried to piece something together. I think it would serve us well to define what we want the API to look like so there's no confusion about things like method names, parameter information, and communication between native code and Flutter.